cmd/containerd-shim-runhcs-v1/service.go

```diff
@@ -42,10 +64,35 @@ type service struct {
 	taskOrPod atomic.Value
 
 	// cl is the create lock. Since each shim MUST only track a single task or
-	// POD. `cl` is used to create the task or POD sandbox. It SHOULD not be
+	// POD. `cl` is used to create the task or POD sandbox. It SHOULD NOT be
 	// taken when creating tasks in a POD sandbox as they can happen
 	// concurrently.
 	cl sync.Mutex
+
+	// shut is closed to signal a shut request is received
+	shut chan struct{}
+	// shutOnce is responsible for closign `shut` and any other necessary cleanup
+	shutOnce sync.Once
+	// GracefulShutdown dictates whether to shutdown gracefully and clean up resources
+	// or exit immediately
+	GracefulShutdown bool
```

Comment:
Does this need to be exported if the struct itself isn't and it's only used within this package?